dnf-behave-tests/features/steps/repo.py

```diff
@@ -200,3 +213,51 @@ def step_countme(context, quantifier):
         | path                     |
         | */metalink.xml?countme=1 |
     """.format(quantifier))
+
+
+@behave.step("I use a {rtype:repo_type} repository based on \"{repo_id}\" with added baseurl \"{baseurl}\"")
+def use_repo_with_baseurl_based_on(context, rtype, repo_id, baseurl):
+    target = "/tmp/" + repo_id + "_with_baseurl"
+    context.execute_steps(
+        """
+        Given I copy directory "{source_repo}" to "{repo_loc}"
+          And I execute bash with args \"createrepo_c --baseurl "{baseurl}" .\" in directory "{loc}"
+        """.format(
+            loc=prepend_installroot(context, target),
+            source_repo=os.path.join(context.dnf.repos_location, repo_id),
+            repo_loc=os.path.join(target, repo_id),
+            baseurl=baseurl))
+
+    if rtype == "local":
+        context.execute_steps(
+            """
+            Given I create and substitute file "/etc/yum.repos.d/{name}.repo" with
+                  \"\"\"
+                  [{name}]
+                  name={name}
+                  baseurl=file://{loc}
+                  enabled=1
+                  gpgcheck=0
+                  \"\"\"
+              And I do not set reposdir
+              And I use the repository "{name}"
+            """.format(
+                loc=prepend_installroot(context, target),
+                name=os.path.basename(target)))
+    else:
+        context.execute_steps(
+            """
+            Given I use the {rtype} repository based on "{loc}"
+            """.format(
+                loc=prepend_installroot(context, target),
+                rtype=rtype))
+
+@behave.step("I make packages from \"{repo}\" repository accessible via {rtype:repo_type}")
+def make_repo_accessible(context, repo, rtype):
+    assert (hasattr(context, 'httpd') or hasattr(context, 'ftpd')), \
+        'Httpd or Ftpd fixture not set. Use @fixture.httpd or @fixture.ftpd tag.'
+    if rtype == "http":
+        host, port = context.httpd.new_http_server(context.dnf.repos_location)
+    elif rtype == "ftp":
+        host, port = context.ftpd.new_ftp_server(context.dnf.repos_location)
+    context.dnf._set(rtype + "-" + repo + "-port", port)
```

Comment:
I think this here is a different approach to storing the ports than what ended up in _"I use the X repository based on Y"_:
```python
context.dnf.ports[rtype + "-" + repo] = port
```